### PR TITLE
produce hints for typo functions and types

### PR DESCRIPTION
**Pull request by danlark1**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Improvement

Short description (up to few sentences):
If a person makes a typo, we would like to warn him and give him correct variants from functions and types. We choose not more than two variants which Levenshtein distance is not greater than two.

### CODE REVIEW

File: dbms/src/Common/IFactoryWithAliases.h [hunk 1]
```diff
@@ -105,6 +106,12 @@ class IFactoryWithAliases
         return aliases.count(name) || case_insensitive_aliases.count(name);
     }
 
+    std::vector<String> getHints(const String & name) const
+    {
+        static const auto registeredNames = getAllRegisteredNames();
```
Comment (alexey-milovidov) [inline dbms/src/Common/IFactoryWithAliases.h]: Variables are in snake_case.

File: dbms/src/Common/NamePrompter.h [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    {
+        DistanceIndexQueue queue;
+        for (size_t i = 0; i < prompting_strings.size(); ++i)
+            appendToQueue(i, name, queue, prompting_strings);
+        return release(queue, prompting_strings);
+    }
+
+private:
+
+    static size_t LevenshteinDistance(const String & lhs, const String & rhs)
```
Comment (alexey-milovidov) [inline dbms/src/Common/NamePrompter.h]: and functions are in camelCase with lowercase initial letter.

Comment (danlark1) [reply dbms/src/Common/NamePrompter.h]: Levenshtein is a name :) I was doubting

**Comment by alexey-milovidov** [comment]
It should be added for all the factories (table functions, storage engines, etc.)

**Pull request merged**